docs/tutorials/HD143006_Part_2.py

```diff
@@ -0,0 +1,468 @@
+gridder = gridding.Gridder(
+    coords=coords,
+    uu=uu,
+    vv=vv,
+    weight=weight,
+    data_re=data.real,  # separating the real and imaginary values of our data
+    data_im=data.imag,
+)
+
+# -
+
+# We now have everything from the last tutorial loaded and can begin the process of optimization and cross validation to improve our image quality.
+#
+# ### Getting the Dirty Image and Creating the Model
+#
+# First, we are going to get the dirty image from our gridder object. We will use the Briggs weighting scale and set `robust=0.0` here as these options lead to an already well optimized image (see [Part 1](HD143006_Part_1.html)).
+
+import torch
+
+img, beam = gridder.get_dirty_image(weighting="briggs", robust=0.0, unit="Jy/arcsec^2")
+# taking the dirty image and making it a tensor
+dirty_image = torch.tensor(img.copy())
+
+# Now we'll create the RML model.
+
+from mpol.precomposed import SimpleNet
+
+model = SimpleNet(coords=coords, nchan=gridder.nchan)
+
+# ## Initializing Model with the Dirty Image
+#
+# We now have our model and data, but before we set out trying to optimize the image we should create a better starting point for our future optimization loops. A good idea for the starting point is the dirty image, since it is already a maximum likelihood fit to the data. The problem with this is that the dirty image contains negative flux pixels, while we impose the requirement that our sources must have all positive flux values. Our solution then is to optimize the RML model to become as close to the dirty image as possible (while retaining image positivity).
+#
+
+
+# To optimize the RML model toward the dirty image, we will create our training loop using a [loss function](../api.html#module-mpol.losses) and an [optimizer](https://pytorch.org/docs/stable/optim.html#module-torch.optim).. MPoL and PyTorch both contain many different optimizers and loss functions, each one suiting different applications. Here we use PyTorch's [mean squared error function](https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html) between the RML model image pixel fluxes and the dirty image pixel fluxes.
+
+optimizer = torch.optim.Adam(model.parameters(), lr=0.5)  # creating the optimizer
+loss_fn = torch.nn.MSELoss()  # creating the MSEloss function from Pytorch
+
+# +
+# %%time
+
+
+for iteration in range(500):
+
+    optimizer.zero_grad()
+
+    model.forward()  # get the predicted model
+    sky_cube = model.icube.sky_cube
+
+    loss = loss_fn(sky_cube, dirty_image)  # calculate the loss
+
+    loss.backward()  # calculate gradients of parameters
+    optimizer.step()  # update the parameters
+# -
+
+# In this tutorial we will be using different methods of RML optimization so we have to save the model, letting us start from this clean starting point each time. Information on saving and loading models and the state_dict can be found [here](https://pytorch.org/tutorials/beginner/saving_loading_models.html).
```

Comment:
What do you mean by different methods of RML optimization? It seems like you're using the same type of optimization, but under different regularization strength settings?